src/Symfony/Component/Workflow/Registry.php

```diff
@@ -56,16 +63,19 @@ public function get($subject, $workflowName = null)
         return $matched;
     }
 
-    private function supports(Workflow $workflow, $className, $subject, $name)
+    private function supports(Workflow $workflow, $supportStrategy, $subject, $workflowName)
     {
-        if (!$subject instanceof $className) {
+        if (is_string($supportStrategy) && !$subject instanceof $supportStrategy) {
+            return false;
+        }
+        if ($supportStrategy instanceof SupportStrategyInterface && !$supportStrategy->supports($workflow, $subject)) {
             return false;
         }
 
-        if (null === $name) {
+        if (null === $workflowName) {
             return true;
         }
 
-        return $name === $workflow->getName();
+        return $workflowName === $workflow->getName();
```

Comment:
So if @stof 's idea above will be implemented, this can even be more simplified, right?